src/content/structured/components/canary/date-input/code.mdx

```diff
@@ -0,0 +1,556 @@
+  label="When would you like to collect your coffee?"
+  disabled="true"
+></ic-date-input>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput label="When would you like to collect your coffee?" disabled />`,
+  },
+];
+
+<ComponentPreview snippets={snippetsDisabled}>
+  <IcDateInput label="When would you like to collect your coffee?" disabled />
+</ComponentPreview>
+
+### Sizes
+
+export const snippetsSizes = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+  size="small"
+></ic-date-input>
+<ic-date-input
+  label="When would you like to collect your coffee?"
+></ic-date-input>
+<ic-date-input
+  label="When would you like to collect your coffee?"
+  size="large"
+></ic-date-input>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput label="When would you like to collect your coffee?" size="small" />
+<IcDateInput label="When would you like to collect your coffee?" />
+<IcDateInput label="When would you like to collect your coffee?" size="large" />`,
+  },
+];
+
+<ComponentPreview
+  snippets={snippetsSizes}
+  style={{
+    display: "flex",
+    flexDirection: "column",
+    gap: "8px",
+    alignItems: "center",
+  }}
+>
+  <IcDateInput
+    label="When would you like to collect your coffee?"
+    size="small"
+  />
+  <IcDateInput label="When would you like to collect your coffee?" />
+  <IcDateInput
+    label="When would you like to collect your coffee?"
+    size="large"
+  />
+</ComponentPreview>
+
+### Custom helper text
+
+export const snippetsCustomHelperText = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+  helper-text="We will have your order ready for you on this date"
+></ic-date-input>`,
+  },
+  {
+    language: "React",
+    snippet: `<IcDateInput
+  label="When would you like to collect your coffee?"
+  helperText="We will have your order ready for you on this date"
+/>`,
+  },
+];
+
+<ComponentPreview snippets={snippetsCustomHelperText}>
+  <IcDateInput
+    label="When would you like to collect your coffee?"
+    helperText="We will have your order ready for you on this date"
+  />
+</ComponentPreview>
+
+### Custom validation
+
+export const snippetsValidation = [
+  {
+    language: "Web components",
+    snippet: `<ic-date-input
+  label="When would you like to collect your coffee?"
+  validation-status="error"
+  validation-text="There is a coffee shortage forecast for that day. Please choose a different date."
+></ic-date-input>
+<ic-date-input
+  label="When would you like to collect your coffee?"
+  validation-status="warning"
+  validation-text="Please be aware that there may be a coffee shortage on that day."
+></ic-date-input>
+<ic-date-input
+  label="When would you like to collect your coffee?"
+  validation-status="success"
+  validation-text="Your coffee will be available for you to collect on this day."
+></ic-date-input>`,
```

Comment:
```suggestion
    snippet: `<ic-date-input label="When would you like to collect your coffee?" validation-status="error" validation-text="There is a coffee shortage forecast for that day. Please choose a different date."></ic-date-input>
<ic-date-input label="When would you like to collect your coffee?" validation-status="warning" validation-text="Please be aware that there may be a coffee shortage on that day."></ic-date-input>
<ic-date-input label="When would you like to collect your coffee?" validation-status="success" validation-text="Your coffee will be available for you to collect on this day."></ic-date-input>`,
```

Reply:
Personally think the original is easier to read when displayed in the code preview